compiler/src/dotty/tools/dotc/transform/PostTyper.scala

```diff
@@ -329,6 +338,8 @@ class PostTyper extends MacroTransform with IdentityDenotTransformer { thisPhase
             )
           }
         case tree: ValDef =>
+          if tree.symbol.is(Inline, butNot = Param) then
```

Comment:
Would this be triggered by
```scala
inline val x = 2
```
? Those things are extremely common, so we should avoid running the inliner just for these.

Reply:
Yes, then we would need to check that the RHS of an `inline val` is a constant after the `Inlining` phase. I try to fit it into another phase.

Reply:
That would cause many more traversals, so yes, good to do this somewhere else.